src/feature-reef/enter_exit.py

```diff
@@ -0,0 +1,41 @@
+# Dunders __enter__ and __exit__
+
+from __future__ import annotations
+
+
+class Context:
+    def fancy_function(self) -> None:
+        print("executing fancy function")
+
+    def __enter__(self) -> Context:
+        print("entering with block")
+
+        return self
+
+    def __exit__(
+            self, exception_type, exception_value,
+            exception_traceback) -> bool:
+        print("exiting with block")
+
+        if (exception_value is not None):
+            print(f"exception '{exception_type}' "
+                  f"occurred at line {exception_traceback.tb_lineno}: "
+                  f"{exception_value}")
+
+            print("ignoring error")
+
+        return True  # Returning true should suppress the exception
+
+
+def main() -> None:
+    with Context() as ctx:  # Initializes ctx as a Context object
+        ctx.fancy_function()
+    # Context.__exit__ gets called
+
+    with Context() as ctx:
+        raise Exception("error")
+    # Context.__exit__ suppresses the exception
```

Comment:
Think these two could be merged?
```suggestion
    with Context() as ctx:  # Initializes ctx as a Context object
        ctx.fancy_function()
        raise Exception("error")
    # Context.__exit__ suppresses the exception
```

Reply:
Sure! I had them separate before since before for a case where an exception was not raised.